### PR TITLE
refactor: URI->URL

### PR DESCRIPTION
**Pull request by Bento007**
## Reason for Change

The goal of this PR is to simplify the numerous download methods used during ingest.

## Changes

- refactor URL-> URI
- add download function to URI

### CODE REVIEW

File: backend/wmg/data/utils.py [hunk 1]
```diff
@@ -149,13 +149,13 @@ def get_collections_from_discover_api():
 
 def get_pinned_ontology_url(name: str):
     """
-    This function retrieves the URL of the pinned ontology based on the provided name.
+    This function retrieves the URI of the pinned ontology based on the provided name.
```
Comment (Bento007) [inline backend/wmg/data/utils.py:152]:
```suggestion
    This function retrieves the URL of the pinned ontology based on the provided name.
```

File: backend/wmg/data/utils.py [hunk 2]
```diff
@@ -149,13 +149,13 @@ def get_collections_from_discover_api():
...
 
     Parameters:
     name (str): The name of the ontology (e.g. cl-basic.obo).
 
     Returns:
-    str: The URL of the pinned ontology.
+    str: The URI of the pinned ontology.
```
Comment (Bento007) [inline backend/wmg/data/utils.py:158]:
```suggestion
    str: The URL of the pinned ontology.
```

File: backend/layers/common/validation.py [hunk 3]
```diff
@@ -73,7 +73,7 @@ def verify_collection_consortia(metadata: Union[CollectionMetadata, CollectionMe
 
 def verify_collection_links(links: List[Link], errors: list) -> None:
     def _error_message(i: int, _url: str) -> dict:
-        return {"name": f"links[{i}]", "reason": "Invalid URL.", "value": _url}
+        return {"name": f"links[{i}]", "reason": "Invalid URI.", "value": _url}
```
Comment (Bento007) [inline backend/layers/common/validation.py:76]:
```suggestion
        return {"name": f"links[{i}]", "reason": "Invalid URL.", "value": _url}
```

File: backend/common/utils/dl_sources/uri.py [hunk 4]
```diff
@@ -162,34 +191,39 @@ def bucket_name(self):
...
 
-def from_url(url: str) -> URL:
-    """Given a URL return a object that can be used by the processing container to download data."""
+def from_url(uri: str) -> typing.Optional[URI]:
+    """Given a URI return a object that can be used by the processing container to download data."""
     for source in RegisteredSources.get():
-        url_obj = source.validate(url)
+        url_obj = source.validate(uri)
         if url_obj:
             return url_obj
```
Comment (nayib-jose-gloria) [inline backend/common/utils/dl_sources/uri.py:225]:
```suggestion
        uri_obj = source.validate(uri)
        if uri_obj:
            return uri_obj
```

**Review by nayib-jose-gloria — COMMENTED** [review]
nit: rename the PR to URL -> URI for clarity, there's a typo

File: backend/common/utils/dl_sources/uri.py [hunk 5]
```diff
@@ -1,33 +1,41 @@
...
+class DownloadFailed(Exception):
+    pass
+
 
 class MissingHeaderException(Exception):
     def __init__(self, detail: str = "", *args, **kwargs) -> None:
         self.detail = "Missing header from response. " + detail
 
 
 # TODO: consider renaming as URI
```
Comment (nayib-jose-gloria) [inline backend/common/utils/dl_sources/uri.py:22]: remove this TODO

File: backend/common/utils/downloader.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+logger = logging.getLogger("processing")
+
+
+def download(
+    uri: str,
+    local_path: str,
+    chunk_size: int = 10 * 2**20,
+) -> None:
+    """
+    Download a file from a url and update the processing_status upload fields in the database
```
Comment (nayib-jose-gloria) [inline backend/common/utils/downloader.py:15]: nit: url vs uri

**Review by nayib-jose-gloria — APPROVED** [review]

**Pull request merged**